der/src/asn1/any.rs

```diff
@@ -282,3 +273,66 @@ impl Tagged for Any {
         self.tag
     }
 }
+
+#[cfg(feature = "alloc")]
+impl ValueOrd for Any {
+    fn value_cmp(&self, other: &Self) -> Result<Ordering> {
+        self.value.der_cmp(&other.value)
+    }
+}
+
+#[cfg(feature = "alloc")]
+impl<'a, T> From<T> for Any
+where
+    T: Into<AnyRef<'a>>,
+{
+    fn from(input: T) -> Any {
+        let anyref: AnyRef<'a> = input.into();
+        Self {
+            tag: anyref.tag(),
+            value: Bytes::new(anyref.value()).expect("invalid ANY"),
+        }
+    }
+}
+
+/// Decode [`AnyRef`] and [`Any`] containers in various formats
+pub trait AnyTransform<'backing, 'any>
+where
+    'backing: 'any,
+    &'any Self: 'any
+        + TryInto<PrintableStringRef<'backing>, Error = Error>
+        + TryInto<VideotexStringRef<'backing>, Error = Error>
+        + TryInto<TeletexStringRef<'backing>, Error = Error>
+        + TryInto<Utf8StringRef<'backing>, Error = Error>
+        + TryInto<Ia5StringRef<'backing>, Error = Error>,
+{
+    /// Attempt to decode an ASN.1 `PrintableString`.
+    fn printable_string(&'any self) -> Result<PrintableStringRef<'backing>> {
+        self.try_into()
+    }
+
+    /// Attempt to decode an ASN.1 `TeletexString`.
+    fn teletex_string(&'any self) -> Result<TeletexStringRef<'backing>> {
+        self.try_into()
+    }
+
+    /// Attempt to decode an ASN.1 `VideotexString`.
+    fn videotex_string(&'any self) -> Result<VideotexStringRef<'backing>> {
+        self.try_into()
+    }
+
+    /// Attempt to decode an ASN.1 `UTF8String`.
+    fn utf8_string(&'any self) -> Result<Utf8StringRef<'backing>> {
+        self.try_into()
+    }
+
+    /// Attempt to decode an ASN.1 `IA5String`.
+    fn ia5_string(&'any self) -> Result<Ia5StringRef<'backing>> {
+        self.try_into()
+    }
+}
+
+impl<'b, 'a: 'b> AnyTransform<'a, 'b> for AnyRef<'a> {}
+
+#[cfg(feature = "alloc")]
+impl<'a> AnyTransform<'a, 'a> for Any {}
```

Comment:
I feel like we could perhaps get rid of this and any kind of method for these and switch to `TryInto`.

Really `Any`/`AnyRef` is something of a legacy crutch used when bootstrapping this library which it would be good to get away from.

It still has value when e.g. dynamically dispatching on a document (e.g. PKCS#8) to figure out how it should be decoded, but it would be nice to avoid other usages whenever possible.

Reply:
the current pkcs8 crate doesn't use any of that does it?
I'd be happy to drop that. I think I just need to rewrite the tests.
I could also keep it with the trait and make the trait `#[deprecated]`

Reply:
These changes are already breaking, so you can modify the `pkcs8` crate accordingly